app/adapters/bagit/persister.rb

```diff
@@ -53,6 +53,8 @@ def valid_lock?(resource)
 
         cached_value = cached_resource[Valkyrie::Persistence::Attributes::OPTIMISTIC_LOCK].first
         cached_value == resource_value
+      rescue Valkyrie::Persistence::ObjectNotFoundError
```

Comment:
Is this basically a check for whether the object is persisted? Is there a way to do that in the body of the method instead of via error handling?

Reply:
We can't check for persisted (the resource was persisted in a different adapter), but I can check for the lock token first.